### PR TITLE
GEODE-7210: Fix RedundancyLevelPart1DUnitTest

### PR DESCRIPTION
**Pull request by igodwin**
Changes made to address flakiness:

- Inline super
- Implement Serializable
- Use Dunit rules DistributedRule and CacheRule
- Use AssertJ
- Use setup and teardown methods
- CountdownLatch rather than Thread.sleep
- No static methods

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by mhansonp** [comment]
Ahem... Spotless.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java [hunk 1]
```diff
@@ -14,78 +14,172 @@
...
+  private static final long TIMEOUT_MILLIS = GeodeAwaitility.getTimeout().getValueInMS();
+  private static final String K1 = "k1";
+  private static final String K2 = "k2";
+  private static final String REGION_NAME = "RedundancyLevelTestBase_region";
+  private static final int CONNECTED_SERVERS = 3;
+  private static final int DEFAULT_SOCKET_READ_TIMEOUT = 3000;
+  private static final int DEFAULT_RETRY_INTERVAL = 10;
+  private static final int DEFAULT_CONNECTED_SERVER_COUNT = 4;
+
+  private static String SERVER0;
```
Comment (kirklund) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: Non-final statics should be capitalized but these should be lowercase.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java [hunk 2]
```diff
@@ -14,78 +14,172 @@
...
+    port1 = vm1.invoke(this::createServerCache);
+    port2 = vm2.invoke(this::createServerCache);
+    port3 = vm3.invoke(this::createServerCache);
+
+    SERVER0 = hostname + port0;
+    SERVER1 = hostname + port1;
+    SERVER2 = hostname + port2;
+    SERVER3 = hostname + port3;
+
+    System.setProperty(GEMFIRE_PREFIX + "bridge.disableShufflingOfEndpoints", "true");
```
Comment (kirklund) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: clearProperty would be more correct:
```
System.clearProperty(GEMFIRE_PREFIX + "bridge.disableShufflingOfEndpoints");
```
But there's even a better way to do this with a JUnit Rule. Delete the clearProperty line and then just add this Rule:
```
@Rule 
public DistributedRestoreSystemProperties restoreSystemProperties = new DistributedRestoreSystemProperties();
```
That will auto-clear any system properties that were set during any test.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java [hunk 3]
```diff
@@ -296,25 +303,243 @@ public void testRedundancySpecifiedNonPrimaryEPFailsDetectionByUnregisterInteres
...
+    cacheServer.setPort(0);
+    cacheServer.start();
+
+    return cacheServer.getPort();
+  }
+
+  private void verifyInterestRegistration() {
+    await().untilAsserted(() -> assertThat(cache.getCacheServers()).hasSize(1));
+
+    CacheServerImpl cacheServer = (CacheServerImpl) cache.getCacheServers().iterator().next();
```
Comment (kirklund) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: Another good thing to change is to move away from direct usage of implementation classes towards usage of interfaces. You'll find that this works:
```
InternalCacheServer cacheServer = (InternalCacheServer) cache.getCacheServers().iterator().next();
```

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java [hunk 4]
```diff
@@ -296,25 +303,243 @@ public void testRedundancySpecifiedNonPrimaryEPFailsDetectionByUnregisterInteres
...
+   * list , then it should be removed from Live Server Map & added to dead server map. It should not
+   * change the current failover set. Failover detection by Register Interest
+   */
+  @Ignore("TODO")
+  @Test
+  public void testRedundancySpecifiedNonFailoverEPFailsDetectionByRegisterInterest() {
+    createClientCache(1, 250, 500);
+    waitConnectedServers();
+    assertThat(pool.getRedundantNames().size()).isEqualTo(1);
+    assertThat(pool.getRedundantNames().contains(SERVER3)).isTrue();
```
[the remark this answers is not in the document]
Comment (mhansonp) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: It passed.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java (hunk 1 above)
Comment (igodwin) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: Done.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java (hunk 2 above)
Comment (igodwin) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: Done, though not in setUp(). Default value for disableShufflingOfEndpoints is false so I removed the "reset" in tearDown().

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java (hunk 3 above)
Comment (igodwin) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: Done. Thank you.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (igodwin) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelPart1DUnitTest.java]: Done. Thank you.

**Pull request closed without merging**